spec.html

```diff
@@ -36148,11 +36050,22 @@ <h1>CreateMapIterator ( _map_, _kind_ )</h1>
         <p>The abstract operation CreateMapIterator takes arguments _map_ and _kind_. This operation is used to create iterator objects for Map methods that return such iterators. It performs the following steps when called:</p>
         <emu-alg>
           1. Perform ? RequireInternalSlot(_map_, [[MapData]]).
-          1. Let _iterator_ be OrdinaryObjectCreate(%MapIteratorPrototype%, &laquo; [[IteratedMap]], [[MapNextIndex]], [[MapIterationKind]] &raquo;).
-          1. Set _iterator_.[[IteratedMap]] to _map_.
-          1. Set _iterator_.[[MapNextIndex]] to 0.
-          1. Set _iterator_.[[MapIterationKind]] to _kind_.
-          1. Return _iterator_.
+          1. Let _closure_ be a new Abstract Closure with no parameters that captures _map_ and _kind_ and performs the following steps when called:
+            1. Let _entries_ be the List that is _map_.[[MapData]].
+            1. Let _index_ be *+0*.
+            1. NOTE: the number of elements of _entries_ mentioned below must be redetermined in each while loop.
```

Comment:
i'm not sure "in each while loop" is the best way to phrase this.

Reply:
the original algorithm says it must be re-evaluated each time so the generator-equivalent is this form imo

Reply:
for one, "the loop" is the totality of all iterations together; perhaps "in each iteration of the loop", but i'm still not sure if that's the best way to phrase it.